docs/en/sql-reference/functions/string-functions.md

```diff
@@ -2539,3 +2539,43 @@ Result:
 │ foo                        │
 └────────────────────────────┘
 ```
+## stringCompare
+
+Compare two strings lexicographically.
+
+**Syntax**
+```sql
+stringComare(str1, str2, str1_off, str2_off, n);
+stringComare(str1, str2)
+```
+
+**Arguments**
+- `str1` — Required, string. One of the strings to be compared.
+- `str2` - Required, string. One of the strings to be compared.
+- `str1_off` — Optional, positive number. The starting position (zero-based index) in `str1` from which the comparison begins.
+- `str2_off` — Optional, positive number. The starting position (zero-based index) in `str2` from which the comparison begins.
+- `n` — The number of bytes to compare in both strings, starting from their respective offsets. `str_off` + `n` could be larger then the length of string, we will align it to the end of the string.
+
+**Returned value**
+- -1 — If `str1`[`str1_off`: `str1_off` + `n`] < `str2`[`str2_off`:`str2_off` + `n`] and `str1_off` < len(`str1`) and `str2_off` < len(`str2`).
+If `str1_off` >= len(`str1`) and `str2_off` < len(`str2`).
+- 0 — If `str1`[`str1_off`: `str1_off` + `n`] = `str2`[`str2_off`:`str2_off` + `n`] and `str1_off` < len(`str1`) and `str2_off` < len(`str2`).
+If `str1_off` >= len(`str1`) and `str2_off` >= len(`str2`).
+- 1 — If `str1`[`str1_off`: `str1_off` + `n`] > `str2`[`str2_off`:`str2_off` + `n`] and `str1_off` < len(`str1`) and `str2_off` < len(`str2`).
+If `str1_off` < len(`str1`) and `str2_off` >= len(`str2`).
+
+
+
+**Example**
+```sql
+SELECT
+    stringCompare('123', '123', 0, 0, 3),
```

Comment:
We should use an English language example here.

Also, it would be cool to show an example with the two argument version and one example with the five argument version.